bin/mail/2016-03-21-teacher-recruitment/generate-ops-workshop-attendees

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env ruby
+require_relative '../mailing-common/mailing-list-utils'
+require 'cdo/geocoder'
+
+# Teachers who attended Ops workshops, minus NYC
+
+teacher_query = <<eos
+  SELECT DISTINCT users.email, users.name, users.current_sign_in_ip FROM users
+    JOIN workshop_attendance ON (users.id = workshop_attendance.teacher_id)
+    JOIN segments ON (segments.id = workshop_attendance.segment_id)
+    JOIN workshops ON (workshops.id = segments.workshop_id)
+    WHERE workshops.location NOT LIKE '%NY%'
```

Comment:
Do we care about efficiency of the query? If so (regardless), I think the following would be more efficient:

SELECT DISTINCT users.email, users.name, users.current_sign_in_ip
FROM (
  SELECT * FROM workshops WHERE workshops.location NOT LIKE '%NY%'
) AS culled_workshops
JOIN segments ON (segments.workshop_id = culled_workshops.id)
JOIN workshop_attendance ON (workshop_attendance.segment_id = segments.id)
JOIN users ON (users.id = workshop_attendance.teacher_id);

Note that the WHERE clause in the original query does not happen until after all the JOINs are resolved.